frontend end (we guess)

Kincso16 opened the pull request.

Commit:
showing how many Q s are left in each survey

Commit:
no refetching at every little change

Lorincz-Gabriella committed:
fixing show question description

Lorincz-Gabriella committed:
Add confetti  after questionnaire completion

Lorincz-Gabriella committed:
fixing celebrate

BHunor-ex commented on FeedBackApp.Frontend/src/pages/dashboards/AdminDashboard.tsx:
```diff
@@ -31,16 +31,15 @@ export default function AdminDashboard() {
-    refetchAdminSurveys();
-  }
-
+  if (user.role !== "Admin") return <Navigate to="/no-access" replace />
```
It's better to write this in two separate lines, because that is more visible.

BHunor-ex commented on FeedBackApp.Frontend/src/pages/dashboards/StudentDashboard.tsx:
```diff
@@ -38,49 +39,52 @@ export default function StudentDashboard() {
+    if (unanswered === 0) {
+      celebrateConfettiRed();
+    }
+  }, [context]);
```
I think `getUnansweredCount` and `celebrateConfettiRed` should also be part of the **dependency array**

Kincso16 replied:
The eslint says "React Hook useEffect has unnecessary dependencies: 'celebrateConfettiRed' and 'getUnansweredCount'. Either exclude them or remove the dependency array. Outer scope values like 'getUnansweredCount' aren't valid dependencies because mutating them doesn't re-render the component.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)"

Commit:
end-points fixed

fixing reviews
organize code structure

Commit:
creating the corresponding azure function endpoint for report compilation and the regarding service layer interface and implementation.

further modifications are awaitable.

# Conflicts:
#	FeedBackApp.Backend/Application/Services/Interfaces/IReportService.cs
#	FeedBackApp.Backend/Application/Services/ReportService.cs

Commit:
Merge remote-tracking branch 'origin/develop' into dynamic-frontend

Commit:
bug fix with teofil on reports generation

SzLRichard committed:
Changed handling of MultiNomialSingleChoiceOther

Commit:
other option fix

Antal-Teofil committed:
adding exception classes

Antal-Teofil committed:
Merge remote-tracking branch 'Explorer-Consulting/dynamic-frontend' into report-endpoint-fix
aha

Antal-Teofil committed:
altering SingleChoice questiontype to adapt to various answer types

Antal-Teofil committed:
- making corresponding function worker skeleton for delivering evaluation reports
- creating custom exception for verifying report compilation

Antal-Teofil committed:
editing AdministratorExcelFormat for showing custom single choice data

Antal-Teofil committed:
revisioning evaluation report compiler

Commit:
Merge remote-tracking branch 'origin/report-endpoint-fix' into dynamic-frontend

Antal-Teofil committed:
fixing single choice other and resolving generation

Commit:
Merge remote-tracking branch 'origin/report-endpoint-fix' into dynamic-frontend

Antal-Teofil committed:
fixing data appearance in pdf generation for question type: SingleChoiceCustom

BHunor-ex commented on FeedBackApp.Backend/Application/Exceptions/ReportCompilationException.cs:
```diff
@@ -0,0 +1,27 @@
+{
+    public class ReportCompilationException : Exception
+    {
+        public string? TemplateId { get; }
```
Why do you need this TemplateId to be property?

Antal-Teofil replied:
Hi, a little misunderstanding has happened and I'm gonna make a new request with time. This exception was several overloads, but u are right this is unnecessary, we are wanted to test whether the template id is of correct format or not, but this exception was already refactored and the property was removed.

Antal-Teofil replied:
*has several overloads

Commit:
Merge remote-tracking branch 'origin/report-endpoint-fix' into dynamic-frontend

BHunor-ex commented on FeedBackApp.Backend/AzureFunctionsAPI/AzureEndPointReaction/Functions/ReportFunctions.cs:
```diff
@@ -12,35 +15,109 @@ public sealed class ReportFunctions(IReportService reportService, ILogger<Report
         private readonly ILogger<ReportFunctions> _reportLogger = logger;
```
So you initialize this _reportLogger property with the logger from the primary constructor, but creating a primary constructor this way will create a reportService and a logger member for the class (so you will have logger and _logger too!), and in the code you sometimes interchanged the two (so you have used _logger which you intended  and logger too , which you didn't). Maybe sometimes you used the reportService also instead of the _reportServices. right now it's  doesn't matter since they are readonly, but imagine if you would do some state management with them then you would have problems  because of it, so either don't use this primary constructor, or watch to don't use the parameter names of it.

Antal-Teofil replied:
Thank u for the remark, that's true, it was also refactored, but unfortunately I wasn't informed early enough about the request, but I will revise it as soon as possible.

BHunor-ex commented on FeedBackApp.Backend/AzureFunctionsAPI/AzureEndPointReaction/Functions/ReportFunctions.cs:
```diff
@@ -12,35 +15,109 @@ public sealed class ReportFunctions(IReportService reportService, ILogger<Report
+
+            if (string.IsNullOrWhiteSpace(templateID))
+            {
+                logger.LogWarning("Empty templateID received in DeliverEvaluationReports.");
```
you used logger instead of _logger

BHunor-ex commented on FeedBackApp.Backend/AzureFunctionsAPI/AzureEndPointReaction/Functions/ReportFunctions.cs:
```diff
@@ -12,35 +15,109 @@ public sealed class ReportFunctions(IReportService reportService, ILogger<Report
+                return bad;
+            }
+
+            logger.LogInformation("DeliverEvaluationReports triggered. templateID={TemplateId}", templateID);
```
here too

BHunor-ex commented on FeedBackApp.Backend/FeedBackApp.Backend.Infrastructure/FeedBackApp.Backend.Infrastructure.csproj:
```diff
@@ -26,6 +26,7 @@
 
   <ItemGroup>
     <Folder Include="Exceptions\" />
+    <Folder Include="Exceptions\" />
```
why is this included twice?

Antal-Teofil replied:
also refactored

BHunor-ex commented on FeedBackApp.Backend/FeedBackApp.Backend.Infrastructure/Persistence/Repository/ReportRepository.cs:
```diff
@@ -33,16 +51,22 @@ public sealed class ReportRepository(AppDBContext context, BlobContainerClient c
+        /// util osztályban. A blob feltöltés <see cref="BinaryData"/>-val történik,
+        /// ami közvetlenül a <c>byte[]</c> dokumentumot használja.</para>
         /// </remarks>
         /// <exception cref="ArgumentException">
```
I think it would be better if these comments (and every comment in the codebase) would be in english instead of hungarian,  but maybe u don't want to deal  with it now

Antal-Teofil replied:
I will do it surely, I made it this way to make the working process smoother for the group members

BHunor-ex commented on FeedBackApp.Backend/FeedBackApp.Core/FeedBackApp.Core.csproj:
```diff
@@ -19,6 +19,7 @@
 
   <ItemGroup>
     <Folder Include="Exceptions\" />
+    <Folder Include="Exceptions\" />
```
Again, it's included two times
